### PR TITLE
[Excel] (Data types) Update JSON schema information

### PR DESCRIPTION
**Pull request by alison-mk**
This PR documents the JSON schema for data types currently in preview. [Work item](https://office.visualstudio.com/OC/_workitems/edit/5624721).

### CODE REVIEW

File: docs/excel/excel-data-types-concepts.md [hunk 1]
```diff
@@ -28,20 +28,23 @@ Use the [`Range.valuesAsJson`](/javascript/api/excel/excel.range#valuesAsJson) p
...
+Each data type uses a JSON metadata schema designed for that type, which defines the [CellValueType](/javascript/api/excel/excel.cellvaluetype) of the data and additional information such as `basicValue`, `numberFormat`, or `address`. Each `CellValueType` has properties available according to that type. For example, the `webImage` type includes the [altText](/javascript/api/excel/excel.webimagecellvalue#altText) and [attribution](/javascript/api/excel/excel.webimagecellvalue#attribution) properties. The following sections show JSON code samples for the formatted number value, entity value, and web image data types.
```
Comment (AlexJerabek) [inline docs/excel/excel-data-types-concepts.md:31]: Consider shortening the first sentence so it's more punchy. Also, I think clarifying the role of the additional information helps the reader conceptualize how all these structures fit together.
```suggestion
Each data type uses a JSON metadata schema designed for that type. This defines the [CellValueType](/javascript/api/excel/excel.cellvaluetype) of the data and additional information about the cell, such as `basicValue`, `numberFormat`, or `address`. Each `CellValueType` has properties available according to that type. For example, the `webImage` type includes the [altText](/javascript/api/excel/excel.webimagecellvalue#altText) and [attribution](/javascript/api/excel/excel.webimagecellvalue#attribution) properties. The following sections show JSON code samples for the formatted number value, entity value, and web image data types.
```

File: docs/excel/excel-data-types-concepts.md [hunk 2]
```diff
@@ -28,20 +28,23 @@ Use the [`Range.valuesAsJson`](/javascript/api/excel/excel.range#valuesAsJson) p
...
+
+The JSON metadata schema for each data type also includes one or more readonly properties that are used when calculations encounter incompatible scenarios, such as a version of Excel that doesn't meet the minimum build number requirement for the data types feature. The property `basicType` is part of the JSON metadata of every data type, and it's always a readonly property.
```
Comment (AlexJerabek) [inline docs/excel/excel-data-types-concepts.md:33]: I think you could use this section to elaborate why `basicType` is useful (it's a fallback when the data type isn't supported or formatted correctly)

File: docs/excel/excel-data-types-concepts.md [hunk 3]
```diff
@@ -28,20 +28,23 @@ Use the [`Range.valuesAsJson`](/javascript/api/excel/excel.range#valuesAsJson) p
...
 
 ## Formatted number values
 
 The [FormattedNumberCellValue](/javascript/api/excel/excel.formattednumbercellvalue) object enables Excel add-ins to define a `numberFormat` property for a value. Once assigned, this number format travels through calculations with the value and can be returned by functions.
 
-The following JSON code sample shows a formatted number value. The `myDate` formatted number value in the code sample displays as **1/16/1990** in the Excel UI.
+The following JSON code sample shows the complete schema of a formatted number value. The `myDate` formatted number value in the code sample displays as **1/16/1990** in the Excel UI. If minimum compatibility requirements aren't met, calculations use the `basicValue` in place of the formatted number.
```
Comment (AlexJerabek) [inline docs/excel/excel-data-types-concepts.md:39]: What are "minimum compatibility requirements" for a formatted number?

File: docs/excel/excel-data-types-concepts.md [hunk 4]
```diff
@@ -65,21 +70,27 @@ const myEntity = {
             type: Excel.CellValueType.string,
             basicValue: "I love llamas."
```
Comment (AlexJerabek) [inline docs/excel/excel-data-types-concepts.md:71]: Getting some serious SimCity flashbacks here ;-)

**Review by AlexJerabek — APPROVED** [review]
Looks good. I have a couple non-blocking questions and comments.

File: docs/excel/excel-data-types-concepts.md (hunk 3 above)
Comment (alison-mk) [reply docs/excel/excel-data-types-concepts.md:39]: This is referencing the minimum compatibility requirements for the data types feature. I'll clarify that.

File: docs/excel/excel-data-types-concepts.md (hunk 4 above)
Comment (alison-mk) [reply docs/excel/excel-data-types-concepts.md:71]: Haha, maybe that's why the engineers included llamas in so many code samples!

**Pull request merged**